File: public/usage-examples/input/mouse_up-1-example.py
from splashkit import *

open_window("Mouse Button Lamp", 800, 600)

while not quit_requested():
    process_events()

    clear_screen(color_white())

    draw_text_no_font_no_size("Hold the left mouse button to turn the lamp red.", color_black(), 20, 20)

    # The lamp stays green while the button is released and turns red for as long as it is held
    if mouse_up(MouseButton.left_button):
        fill_circle(color_green(), 400, 330, 110)
        draw_text_no_font_no_size("Left button: up", color_black(), 20, 60)
    else:
        fill_circle(color_red(), 400, 330, 110)
        draw_text_no_font_no_size("Left button: down", color_black(), 20, 60)

    draw_circle(color_black(), 400, 330, 110)

    refresh_screen_with_target_fps(60)

close_all_windows()

File: public/usage-examples/input/mouse_up-1-example.cpp
#include "splashkit.h"

int main()
{
    open_window("Mouse Button Lamp", 800, 600);

    while (!quit_requested())
    {
        process_events();

        clear_screen(COLOR_WHITE);

        draw_text("Hold the left mouse button to turn the lamp red.", COLOR_BLACK, 20, 20);

        // The lamp stays green while the button is released and turns red for as long as it is held
        if (mouse_up(LEFT_BUTTON))
        {
            fill_circle(COLOR_GREEN, 400, 330, 110);
            draw_text("Left button: up", COLOR_BLACK, 20, 60);
        }
        else
        {
            fill_circle(COLOR_RED, 400, 330, 110);
            draw_text("Left button: down", COLOR_BLACK, 20, 60);
        }

        draw_circle(COLOR_BLACK, 400, 330, 110);

        refresh_screen(60);
    }

    close_all_windows();

    return 0;
}
